### PR TITLE
concourse pr_pipeline: Switch to building centos7; Only master PRs

### PR DESCRIPTION
**Pull request by dsharp-pivotal**
Authored-by: David Sharp <dsharp@pivotal.io>

A test pipeline (with the pr 'put' steps removed) is up here: https://k8s.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_pr/

### CODE REVIEW

**Comment by dsharp-pivotal** [comment]
I think this is missing something: PRs are failing ICW b/c gp_inject_fault() is missing. Please let me know if there's an easy fix for that.

> +ERROR:  could not open extension control file "/usr/local/greenplum-db-devel/share/postgresql/extension/gp_inject_fault.control": No such file or directory

**Comment by berlin-ab** [comment]
> I think this is missing something: PRs are failing ICW b/c gp_inject_fault() is missing. Please let me know if there's an easy fix for that.
> 
> > +ERROR:  could not open extension control file "/usr/local/greenplum-db-devel/share/postgresql/extension/gp_inject_fault.control": No such file or directory

Probably need to configure with `--enable-debug-extensions`, but I'd have to look closer to be sure.

File: concourse/pipelines/pr_pipeline.yml [hunk 1]
```diff
@@ -3,83 +3,155 @@
...
+      file: gpdb_pr/concourse/tasks/sync_tools.yml
+      image: gpdb6-centos7-build
+      params:
+        TARGET_OS: centos
+        TARGET_OS_VERSION: 7
+    - task: compile_gpdb_centos7
+      file: gpdb_pr/concourse/tasks/compile_gpdb.yml
+      image: gpdb6-centos7-build
+      params:
+        CONFIGURE_FLAGS: {{configure_flags}}
```
Comment (chrishajas) [inline concourse/pipelines/pr_pipeline.yml]: I think changing `configure_flags` to `configure_flags_with_extensions` should fix the issue.

Comment (dsharp-pivotal) [reply concourse/pipelines/pr_pipeline.yml]: configure_flags_with_extensions was not part of the pr_pipeline's secrets, but we just added it.

**Comment by dsharp-pivotal** [comment]
Okay, things are looking better now. The most recent PR build on the test pipeline failed, but it wasn't the pipeline's fault! ;)

PTAL

**Pull request merged**